test/browser/newsletter-signup.test.jsx

```diff
@@ -0,0 +1,65 @@
+var NewsletterSignupForm = require('../../components/newsletter-signup/SignupForm.jsx');
+var validateSignupForm = require('../../components/newsletter-signup/validateSignupForm.js');
+
+var stubContext = require('./stub-context.jsx');
+var Util = require('../util.js');
+
+var GOOD_EMAIL_EXAMPLES = [
+  "hello@example.com",
+  "hello123@example.com",
+  "hello+test@example.com",
+  "hello123+test@example.com",
+  "hello!@example.com",
+  "!@example.com",
+  "#!$%&'*+-/=?^_`{}|~@example.org"
+];
+
+var BAD_EMAIL_EXAMPLES = [
+  "",
+  "h.e.llo",
+  "h@e@llo@example.com",
+  "@example.com",
+  "@example.com.",
+  "h..ello@examplecom",
+  "hello@example..com",
+  "hel lo@example.com",
+  "@.com",
+  "@",
+  "hello",
+  ".com",
+  "."
+];
```

Comment:
@stephaniemcv thanks! 👯 

@Pomax asked Salesforce on Twitter about this yesterday and they replied! https://twitter.com/TheRealPomax/status/747560536015941633 I'm gonna use the REGEX they suggested so that we can *blame* them if anything goes wrong 😄 😁